foyer/forcefield.py

```diff
@@ -118,6 +118,53 @@ def _topology_from_parmed(structure, non_element_types):
     positions = structure.positions
     return topology, positions
 
+
+def _topology_from_residue(res):
+    """Converts a openmm.app.Topology.Residue to openmm.app.Topology"""
+    topology = app.Topology()
+    chain = topology.addChain()
+    new_res = topology.addResidue(res.name, chain)
+
+    atoms = dict()  # omm.Atom in res : omm.Atom in topology
+
+    for res_atom in res.atoms():
+        topology_atom = topology.addAtom(name=res_atom.name,
+                         #element=elem.Element.getBySymbol(res_atom.name),
+                         element=res_atom.element,
```

Comment:
This line may cause some problems when trying to atomtype coarse-grained systems (where the convention for Foyer is to prepend "element" names with an underscore).  This could be tested with a box of united-atom alkanes since the TraPPE forcefield is already contained in Foyer.  If it turns out there's not a good way around this, we could have `createSystem` check to see if the system contains any coarse-grained particles, in which case the old approach to atomtyping could be used.